foundation/foundation-upgrade/src/main/java/com/boozallen/aissemble/upgrade/migration/v1_11_0/HiveUniversalConfigYAMLMigration.java

```diff
@@ -0,0 +1,110 @@
+    private static final List<String> mysqlAuthForConfigStore = Arrays.asList("username");
+    private static final HashSet<String> linesToRemove = new HashSet<>(Arrays.asList("username: hive", "- name: javax.jdo.option.ConnectionUserName" ,"value: hive", "description: Username to use against metastore database", "# NB: UCS"));
+    private static final String DEFAULT_HIVE_USERNAME_AND_PASSWORD = "hive";
+    @Override
+    protected boolean shouldExecuteOnFile(File file) {
+        try {
+            YamlUtils.YamlObject yaml = YamlUtils.loadYaml(file);
+            if(yaml.hasObject("aissemble-hive-metastore-service-chart"))
+            {
+                yaml = yaml.getObject("aissemble-hive-metastore-service-chart");
+            }else{
+                return false;
+            }
+            if (yaml.hasObject("mysql", "auth")) {
+                YamlUtils.YamlObject mysqlAuth = yaml.getObject("mysql", "auth");
+                for(String key : mysqlAuth.keySet())
+                {
+                    if(mysqlAuthForConfigStore.contains(key) && mysqlAuth.get(key).equals(DEFAULT_HIVE_USERNAME_AND_PASSWORD))
+                    {
+                        return true;
+                    }
+                }
+
+            }
+
+            if (yaml.hasList("configMap", "metastoreServiceConfig", "properties")) {
+                List<YamlUtils.YamlObject> properties = yaml.getListOfObjects("configMap", "metastoreServiceConfig", "properties");
+                for(YamlUtils.YamlObject property : properties)
+                {
+                    if(propertiesForConfigStore.contains(property.get("name")) && property.get("value").equals(DEFAULT_HIVE_USERNAME_AND_PASSWORD))
+                    {
+                        return true;
+                    }
+                   }
+            }
+            return false;
+
+        } catch (YAMLException e) {
+            if (logger.isDebugEnabled()) {
+                logger.debug("Failed to parse values yaml file: {}", file.getPath(), e);
+            } else {
+                logger.info("Failed to parse values yaml file: {}", file.getName());
+            }
+        }
+        catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+        return false;
+    }
+
+    @Override
+    protected boolean performMigration(File file) {
+        try {
+            File tempFile = new File("tempValuesFile.yaml");
+            BufferedWriter writer = new BufferedWriter(new FileWriter(tempFile));
+            List<String> lines = Files.readAllLines(file.toPath());
+            for (int i = 0; i < lines.size(); i++) {
+                String line = lines.get(i).trim();
+                boolean skip = true;
+                if(i >= 1 && lines.get(i-1).trim().equals("- name: javax.jdo.option.ConnectionPassword"))
+                {
+                    skip = false;
+                }
```

Comment:
Why do we need this skip logic? Seems the `linesToRemove` below would be sufficient.

Reply:
Yup! Since ConnectionPassword's value also have same value of ConnectionUsername, without this logic, password's value will be wiped out. 


i.e. 

```
      <property>
        <name>javax.jdo.option.ConnectionUserName</name>
        <value>hive</value>
        <description>Username to use against metastore database</description>
      </property>
      <property>
        <name>javax.jdo.option.ConnectionPassword</name>
        <value>hive</value>
        <description>Password to use against metastore database</description>
      </property>
```

Reply:
Let's make `skip` be `skipConnectionPassword` then so it's easier to understand and default that to `false`.